Cargo.toml

```diff
@@ -14,6 +14,7 @@ async-trait = "0.1.40"
 lazy_static = "1.4.0"
 pin-project = "0.4.23"
 futures = "0.3.5"
+futures-locks = "0.6.0"
```

Comment:
I think this can be removed as almost all locks were removed from context and curried futures